### ISSUE TITLE
Enhance navigation of `ItemList`

### ISSUE THREAD

**Issue by feois** [godotengine/godot-proposals#6798]
### Describe the project you are working on

Not important

### Describe the problem or limitation you are having in your project

When the selected item is at the start or the end of the row, event `ui_left` or `ui_right` respectively does nothing

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

See below

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

When invoking event `ui_right` while the selected item is at the end of the row of `ItemList`, it should switch selection to the first item of next row, vice versa, when invoking event `ui_left` while the selected item is at the front of the row, selection should switch to the last item of previous row.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Well yes, this can be implemented in scripts, it's inconvenient (plus you need to check if `ItemList` is the focus)

### Is there a reason why this should be core and not an add-on in the asset library?

`ItemList` is core node

**Comment by feois** [comment]
While reading `ItemList` source code, I found out that it's not a limitation but instead an intentional choice (`!IS_SAME_ROW`).
If so, I propose that it's made a choice.

**Comment by Calinou** [comment]
This can likely be done by adding an **Allow Cycle** boolean property, similar to what I proposed in https://github.com/godotengine/godot-proposals/issues/5139.

**Comment by Kazuo256** [comment]
I agree with Calinou, and add that wraparound should also be available for vertical movement (with `ui_down` and `ui_up`). Whether it stays on the same row/column or moves on to the next should maybe be a variation. For instance, there could be two "enum" properties:

* Vertical wraparound (None, Same column, Next column)
* Horizontal wraparound (None, Same row, Next row)
